tests/sysversioninfo.pyi

```diff
@@ -29,3 +29,27 @@ if sys.version_info <= (3, 0): ...  # Y006 Use only < and >= for version compari
 if sys.version_info < (3, 5): ...
 if sys.version_info >= (3, 5): ...
 if (2, 7) <= sys.version_info < (3, 5): ...  # Y002 If test must be a simple comparison against sys.platform or sys.version_info
+
+if sys.version_info >= (3, 10):
+    def foo1(x, *, bar=True, baz=False): ...
+elif sys.version_info >= (3, 9):
+    def foo1(x, *, bar=True): ...
+else:
+    def foo1(x): ...
+
+if sys.version_info < (3, 9):
+    def foo2(x): ...
+elif sys.version_info < (3, 10):
+    def foo2(x, *, bar=True): ...
+
+if sys.version_info < (3, 10):  # Y066 When using if/else with sys.version_info, put the code for new Python versions first, e.g. "if sys.version_info >= (3, 10)"
+    def foo3(x): ...
+else:
+    def foo3(x, *, bar=True): ...
+
+if sys.version_info < (3, 9):
+    def foo4(x): ...
+elif sys.version_info < (3, 10):  # Y066 When using if/else with sys.version_info, put the code for new Python versions first, e.g. "if sys.version_info >= (3, 10)"
+    def foo4(x, *, bar=True): ...
+else:
+    def foo4(x, *, bar=True, baz=False): ...
```

Comment:
I'm glad I asked you to make the test cases more realistic ;) because I think it makes it more obvious that it's slightly _odd_ that we'd emit an error on the penultimate `elif` of an `if/elif` chain that ends in `else`, but _not_ on any of the preceding `if`s or `elif`s.

Maybe we should also emit an error on the first `if` condition of this chain as well? I think we could detect whether a chain eventually ends with an `else` by using a function like this. (We want to emit the error on all problematic `if`s if they're part of a chain that eventually ends in an `else`, I think):

```pycon
>>> import ast
>>> def if_chain_ends_with_else(if_chain: ast.If) -> bool:
...     orelse = if_chain.orelse
...     if len(orelse) == 1 and isinstance(orelse[0], ast.If):
...         return if_chain_ends_with_else(orelse[0])
...     return bool(orelse)
... 
>>> source = '''\
... if a:
...     ...
... elif b:
...     ...
... elif c:
...     ...
... else:
...     ...
... '''
>>> if_ = ast.parse(source).body[0]
>>> if_
<ast.If object at 0x10265e8d0>
>>> if_chain_ends_with_else(if_)
True
>>> source = '''\
... if a:
...     ...
... elif b:
...     ...
... '''
>>> if_chain_ends_with_else(ast.parse(source).body[0])
False
>>> source = '''\
... if a:
...     ...
... else:
...     ...
... '''
>>> if_chain_ends_with_else(ast.parse(source).body[0])
True
```

Reply:
I hope I understood correctly - I modified the code so that it emits the error for all `if`/`elif` blocks in the chain (each error references its own `version_info` number)